### ISSUE TITLE
[Bug]: Controlnet is not working

### ISSUE THREAD

**Issue by syker-uk** [Mikubill/sd-webui-controlnet#2221]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Controlnet doesn't modify the generation; I still get the same images than with vanilla SD.
All the models are loaded in the webui, controlnet is hooked, generating images doesn't throw exceptions. Any other extension works, the problem is specific to controlnet. 

### Steps to reproduce the problem

Followed the instruction as stated in the README.md. 
1. git cloned into the extensions folder
2. updated through webui 
3. downloaded models with this handy script: 
```
#!/bin/bash

links=(
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-canny-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-color-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-depth-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-fuser-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-sketch-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/coadapter-style-sd15v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_canny_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_canny_sd15v2.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_color_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_depth_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_depth_sd15v2.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_keypose_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_openpose_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_seg_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_sketch_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_sketch_sd15v2.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_style_sd14v1.pth
https://huggingface.co/TencentARC/T2I-Adapter/resolve/main/models/t2iadapter_zoedepth_sd15v1.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_canny.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_depth.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_hed.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_mlsd.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_normal.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_openpose.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_scribble.pth
https://huggingface.co/lllyasviel/ControlNet/resolve/main/models/control_sd15_seg.pth
)

for i in ${links[@]}; do
    wget $i
done
```
4. restarted everything (machine, notebook, ui) 

### Things I've tried: 
Browser: [Firefox, Chrome] Vanilla Chrome, no addons, no ad blocker
LowRam: [Enabled, Disabled]
Image: not using an image does return an error so the reference image is definitely loaded 
Flags: [--disable-gradio-queue, --no-half, and all the other stuff I found in open and closed issues] 
Re-install, re-download models.
Preprocessor: [openpose, openpose_full, dw_openpose_full] 
Model: [ control_sd15_openpose [fef5e48e], t2iadapter_openpose_sd14v1 [7e267e5e] ]
Disabled all other extensions.
Removed webui_config.json and cache.json and restarted.

### What should have happened?

Controlnet should control the generation. 

### Command Line Arguments

```Shell
!python launch.py --share --skip-version-check --ckpt ../../datasets/stable-diffusion-classic/SDv1.5.ckpt
```

### List of enabled extensions

I disabled all other extension, restarted the notebook and than the server. 

### Console logs

```Shell
No error in log: 

/notebooks/stable-diffusion-webui
Python 3.9.16 (main, Dec  7 2022, 01:11:51) 
[GCC 9.4.0]
Version: v1.6.0-2-g4afaaf8a
Commit hash: 4afaaf8a020c1df457bcf7250cb1c7f609699fa7
Installing clip
Installing open_clip
Installing requirements for CodeFormer
Installing requirements
Installing sd-webui-controlnet requirement: mediapipe
Installing sd-webui-controlnet requirement: svglib
Installing sd-webui-controlnet requirement: fvcore
Installing sd-webui-controlnet requirement: changing opencv-python version from 4.6.0.66 to 4.8.0
Launching Web UI with arguments: --share --skip-version-check --ckpt ../../datasets/stable-diffusion-classic/SDv1.5.ckpt
no module 'xformers'. Processing without...
No SDP backend available, likely because you are running in pytorch versions < 2.0. In fact, you are using PyTorch 1.12.1+cu116. You might want to consider upgrading.
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
2023-11-04 08:13:01,406 - ControlNet - INFO - ControlNet v1.1.410 for AnimateDiff
ControlNet preprocessor location: /notebooks/stable-diffusion-webui/extensions/sd-webui-controlnet-animatediff/annotator/downloads
2023-11-04 08:13:02,540 - ControlNet - INFO - ControlNet v1.1.410 for AnimateDiff
Loading weights [4c86efd062] from ../../datasets/stable-diffusion-classic/SDv1.5.ckpt
Running on local URL:  http://127.0.0.1:7860
Running on public URL: https://2384acfa9807120363.gradio.live

This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)
Startup time: 101.0s (prepare environment: 80.6s, import torch: 4.5s, import gradio: 1.2s, setup paths: 0.9s, initialize shared: 0.5s, other imports: 1.0s, setup codeformer: 0.5s, load scripts: 2.4s, create ui: 0.6s, gradio launch: 8.7s).
Creating model from config: /notebooks/stable-diffusion-webui/configs/v1-inference.yaml
Downloading (…)olve/main/vocab.json: 100%|███| 961k/961k [00:00<00:00, 41.4MB/s]
Downloading (…)olve/main/merges.txt: 100%|███| 525k/525k [00:00<00:00, 24.8MB/s]
Downloading (…)cial_tokens_map.json: 100%|██████| 389/389 [00:00<00:00, 172kB/s]
Downloading (…)okenizer_config.json: 100%|██████| 905/905 [00:00<00:00, 380kB/s]
Downloading (…)lve/main/config.json: 100%|██| 4.52k/4.52k [00:00<00:00, 692kB/s]
Applying attention optimization: Doggettx... done.
Model loaded in 34.7s (load weights from disk: 31.2s, create model: 0.8s, apply weights to model: 2.1s, calculate empty prompt: 0.4s).
2023-11-04 08:19:49,149 - ControlNet - INFO - Loading model: t2iadapter_openpose_sd14v1 [7e267e5e]
2023-11-04 08:19:49,887 - ControlNet - INFO - Loaded state_dict from [/notebooks/stable-diffusion-webui/extensions/sd-webui-controlnet-animatediff/models/t2iadapter_openpose_sd14v1.pth]
2023-11-04 08:19:49,888 - ControlNet - INFO - t2i_adapter_config
2023-11-04 08:19:50,635 - ControlNet - INFO - ControlNet model t2iadapter_openpose_sd14v1 [7e267e5e] loaded.
2023-11-04 08:19:50,647 - ControlNet - INFO - Loading preprocessor: openpose_full
2023-11-04 08:19:50,647 - ControlNet - INFO - preprocessor resolution = 512
2023-11-04 08:19:55,307 - ControlNet - INFO - ControlNet Hooked - Time = 6.484411001205444
  0%|                                                    | 0/20 [00:00<?, ?it/s]
  5%|██▏                                         | 1/20 [00:01<00:24,  1.27s/it]
 10%|████▍                                       | 2/20 [00:02<00:21,  1.22s/it]
 15%|██████▌                                     | 3/20 [00:03<00:20,  1.19s/it]
 20%|████████▊                                   | 4/20 [00:04<00:18,  1.18s/it]
 25%|███████████                                 | 5/20 [00:05<00:17,  1.17s/it]
 30%|█████████████▏                              | 6/20 [00:07<00:16,  1.16s/it]
 35%|███████████████▍                            | 7/20 [00:08<00:15,  1.16s/it]
 40%|█████████████████▌                          | 8/20 [00:09<00:13,  1.16s/it]
 45%|███████████████████▊                        | 9/20 [00:10<00:12,  1.16s/it]
 50%|█████████████████████▌                     | 10/20 [00:11<00:11,  1.16s/it]
 55%|███████████████████████▋                   | 11/20 [00:12<00:10,  1.16s/it]
 60%|█████████████████████████▊                 | 12/20 [00:14<00:09,  1.16s/it]
 65%|███████████████████████████▉               | 13/20 [00:15<00:08,  1.16s/it]
 70%|██████████████████████████████             | 14/20 [00:16<00:06,  1.16s/it]
 75%|████████████████████████████████▎          | 15/20 [00:17<00:05,  1.16s/it]
 80%|██████████████████████████████████▍        | 16/20 [00:18<00:04,  1.16s/it]
 85%|████████████████████████████████████▌      | 17/20 [00:19<00:03,  1.16s/it]
 90%|██████████████████████████████████████▋    | 18/20 [00:20<00:02,  1.16s/it]
 95%|████████████████████████████████████████▊  | 19/20 [00:22<00:01,  1.16s/it]
100%|███████████████████████████████████████████| 20/20 [00:23<00:00,  1.16s/it]

Total progress: 100%|███████████████████████████| 20/20 [00:23<00:00,  1.16s/it]
```

### Additional information

I'm running this in paperspace pro.

**Comment by sdbds** [comment]
It seems you try to use T2i SD1.4 model.
Can you choose other CNet model like control_sd15_canny.pth?

**Comment by syker-uk** [comment]
This is the exhaustive list of things I tried:
```
Browser: [Firefox, Chrome] Vanilla Chrome, no addons, no ad blocker
LowRam: [Enabled, Disabled]
Image: not using an image does return an error so the reference image is definitely loaded
Flags: [--disable-gradio-queue, --no-half, and all the other stuff I found in open and closed issues]
Re-install, re-download models.
Preprocessor: [openpose, openpose_full, dw_openpose_full]
Model: [ control_sd15_openpose [fef5e48e], t2iadapter_openpose_sd14v1 [7e267e5e] ]
Disabled all other extensions.
Removed webui_config.json and cache.json and restarted.
```
And now you can add control_sd15_canny.pth t the list of models I tried.

**Comment by sdbds** [comment]
> You can add control_sd15_canny.pth to the list of models I tried. Still, controlnet doesn't do anything. I've tried with your repo and with the ControlNet for animediff (which is the output you see here). All the same.

from log, i found you used python is 3.9.16.We recommend python over 3.10

**Comment by syker-uk** [comment]
Changing the python version did the trick and it was an easy fix. To anyone on paperspace: when you create the base Stable Diffusion notebook, go to advanced setting and paste `cyberes/gradient-base-py3.10:latest` in the container name, change the model path at launch to `/datasets/stable-diffusion-classic/SDv1.5.ckpt`and it should work.
